Prog/Crimes_forweb.sas

```diff
@@ -0,0 +1,86 @@
+/**************************************************************************
+ Program:  Crime_forweb
+ Library:  Police
+ Project:  NeighborhoodInfo DC
+ Author:   Rob Pitingolo
+ Created:  11/07/2017
+ Version:  SAS 9.4
+ Environment:  Windows
+ Modifications: 
+
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas"; 
+
+** Define libraries **;
+%DCData_lib( Police )
+%DCData_lib( Web )
+
+
+/***** Update the let statements for the data you want to create CSV files for *****/
+
+%let library = police; /* Library of the summary data to be transposed */
+%let outfolder = crime; /* Name of folder where output CSV will be saved */
+%let sumdata = crimes_sum; /* Summary dataset name (without geo suffix) */
+%let start = 2000; /* Start year */
+%let end = 2016; /* End year */
+%let keepvars = crimes_pt1 Crimes_pt1_property Crimes_pt1_violent Crime_rate_pop; /* Summary variables to keep and transpose */
+
+
+/***** Update the web_varcreate marcro if you need to create final indicators for the website after transposing *****/
+
+%macro web_varcreate;
+
+Crimes_pt1_per1000 = crimes_pt1 / Crime_rate_pop*1000;
+Crimes_pt1_property_per1000 = Crimes_pt1_property / Crime_rate_pop*1000;
+Crimes_pt1_violent_per1000 = Crimes_pt1_violent / Crime_rate_pop*1000;
+
+%mend web_varcreate;
+
+
+
+/**************** DO NOT UPDATE BELOW THIS LINE ****************/
+
+%macro csv_create(geo);
+			 
+%web_transpose(&library., &outfolder., &sumdata., &geo., &start., &end., &keepvars. );
```

Comment:
Thanks, Peter. I'll review these suggested edits. As a general point, I don' think we will ever have to run ACS data through %web_transpose() because each ACS is already just it's own year? Please let me know if I'm not thinking about that correctly.

Reply:
@rpitingolo Yes, I see. I hadn't thought of that but it makes sense. OK, if we assume that all the variables for that macro will be 4-digit years, that's fine.

Reply:
@ptatian NIDC summary files are all year-based, as far as I can tell, and since we use those files to create the profile indicators I think we'll always have 4-digit years.

Reply:
@rpitingolo Got it. I still think you should add the _timeframe_ var to the macro _by_ statement, however.